contrib/garyburd/redigo/redigo.go

```diff
@@ -127,6 +127,10 @@ func (tc Conn) Do(commandName string, args ...interface{}) (reply interface{}, e
 		}
 	}
 
+	if ctx == nil {
+		ctx = context.Background()
+	}
```

Comment:
We should initialize `ctx` in the `var` declaration instead.

Reply:
If we do that and the caller pass a nil context as last argument, the application will panic.
But it's maybe the responsibility of the caller to pass a valid context.

Reply:
Yes, I think that's the responsibility of the caller.
I think if the caller passes in an invalid context, we don't want to substitute it. Rather, we should just pass it on.

Reply:
:+1: Fixed !